packages/composites/augur/src/dataProviders/sportsdataio.ts

```diff
@@ -0,0 +1,153 @@
+import { Requester, Validator } from '@chainlink/ea-bootstrap'
+import { Execute } from '@chainlink/types'
+import * as Sportsdataio from '@chainlink/sportsdataio-adapter'
+import { BigNumber } from 'ethers'
+import { packCreation } from '../methods/createMarkets'
+import { packResolution } from '../methods/resolveMarkets'
+
+interface NFLEvent {
+  PointSpread: number
+  Date: string
+  GlobalGameID: number
+  GlobalAwayTeamID: number
+  GlobalHomeTeamID: number
+  Status: string
+}
+
+type SportsdataioNFLSchedule = NFLEvent[]
+
+const getNFLSchedule = async (id: string, season: string, sportsdataExec: Execute): Promise<SportsdataioNFLSchedule> => {
+  const input = {
+    id,
+    data: {
+      sport: 'nfl',
+      endpoint: 'schedule',
+      season
+    }
+  }
+  const response = await sportsdataExec(input)
+  return response.result
+}
+
+const getNFLScores = async (id: string, season: string, sportsdataExec: Execute): Promise<NFLScores[]> => {
+  const input = {
+    id,
+    data: {
+      sport: 'nfl',
+      endpoint: 'scores',
+      season
+    }
+  }
+  const response = await sportsdataExec(input)
+  return response.result
+}
+
+const getSeason = () => `${new Date().getFullYear()}REG` // TODO: Sufficient? What if at the end of the year?
+
+const createParams = {
+  sport: true,
+  daysInAdvance: true,
+  startBuffer: true,
+  contract: true,
+}
+
+export const create: Execute = async (input) => {
+  const validator = new Validator(input, createParams)
+  if (validator.error) throw validator.error
+
+  const sport = validator.validated.data.sport
+  if (sport.toUpperCase() !== 'NFL') {
+    throw Error(`Unknown sport for Sportsdataio: ${sport}`)
+  }
+
+  const daysInAdvance = validator.validated.data.daysInAdvance
+  const startBuffer = validator.validated.data.startBuffer
+  // TODO: Doesn't do anything yet
+  // const contract: ethers.Contract = validator.validated.data.contract
+
+  const sportsdataioExec = Sportsdataio.makeExecute()
+
+  const schedule = await getNFLSchedule(input.id, getSeason(), sportsdataioExec)
+
+  // filter markets and build payloads for market creation
+  const packed = [];
+  for (const event of schedule) {
+    const startTime = Date.parse(event.Date)
+    const diffTime = startTime - Date.now()
+    if (diffTime / 1000 < startBuffer) continue
+    if (diffTime / (1000 * 3600 * 24) > daysInAdvance) continue
+
+    // TODO: Doesn't do anything right now
+    /*const [headToHeadMarket, spreadMarket, totalScoreMarket]: [ethers.BigNumber, ethers.BigNumber, ethers.BigNumber]
+      = await contract.getEventMarkets(event.GlobalGameID)*/
+
+    const start = Date.parse(event.Date) / 1000
+    // TODO: Missing homeSpread
+    // TODO: Missing totalScore
```

Comment:
from Yaser: 
here is how we want to handle the value for totalScore. When Status = Final, we can add value:
`let’s please add it as AwayScore + HomeScore when Status = Final`

Reply:
This has been resolved offline (this case will never happen as we don't create markets when Status = Final)